api/src/test/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/RoutineCalculationTest.java

```diff
@@ -0,0 +1,312 @@
+package org.openmrs.module.eptsreports.reporting.calculation.pvls;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyList;
+import static org.mockito.Matchers.anyMapOf;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Matchers.isNull;
+import static org.mockito.Mockito.when;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.openmrs.Concept;
+import org.openmrs.EncounterType;
+import org.openmrs.Location;
+import org.openmrs.Obs;
+import org.openmrs.Patient;
+import org.openmrs.Person;
+import org.openmrs.calculation.patient.PatientCalculationContext;
+import org.openmrs.calculation.patient.PatientCalculationServiceImpl;
+import org.openmrs.calculation.result.CalculationResultMap;
+import org.openmrs.calculation.result.ListResult;
+import org.openmrs.calculation.result.ObsResult;
+import org.openmrs.calculation.result.SimpleResult;
+import org.openmrs.module.eptsreports.metadata.HivMetadata;
+import org.openmrs.module.eptsreports.reporting.calculation.BooleanResult;
+import org.openmrs.module.eptsreports.reporting.calculation.EptsCalculations;
+import org.openmrs.module.eptsreports.reporting.utils.EptsReportConstants.PatientsOnRoutineEnum;
+import org.openmrs.module.reporting.common.TimeQualifier;
+import org.openmrs.test.BaseContextMockTest;
+
+public class RoutineCalculationTest extends BaseContextMockTest {
```

Comment:
can we move testing these just fixed criterias into https://github.com/esaude/openmrs-module-eptsreports/blob/master/api/src/test/java/org/openmrs/module/eptsreports/reporting/intergrated/calculation/pvls/RoutineCalculationTest.java to be run as integrated tests rather. it will even make is easy to review them.

Reply:
I don't know. The task says unit test. How will it make it easy to review?

Reply:
Lets not replace the unit tests, but yes please extend the Routine integration tests by adding scenarios to cater for criteria2 & 3